### PR TITLE
bspwm: various improvements

### PR DESCRIPTION
**Pull request by ncfavier**
### Description

- Allows the user to reload their config on-the-fly by exposing the configuration file to `~/.config/bspwm/bspwmrc`
- Fixes some shell escaping issues using `lib.strings.escapeShellArg`
- Style improvements
- Adds myself as a maintainer since this module doesn't have one. @Valodim, is this OK with you?

This has been tested with a reasonably general config.

### Checklist

- [X] Change is backwards compatible.

- [X] Code formatted with `./format`.

- [X] Code tested through `nix-shell --pure tests -A run.all`.

- [ ] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [X] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/doc/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

### CODE REVIEW

**Comment by Valodim** [comment]
Sure :)

Improvements lgtm :+1: 

I don't know if the `~/.config/bspwm/bspwmrc` part is acceptable, in the end it's a tradeoff of purity vs reloadability, I don't know which weighs stronger here. I'm good with either variant, will leave that up to @rycee

**Comment by ncfavier** [comment]
We're not losing any purity, that file is still read-only. It's just available at a known location.

Compare with https://github.com/nix-community/home-manager/blob/master/modules/services/sxhkd.nix#L70-L73

**Comment by Valodim** [comment]
sounds good, and aligning bspwm with sxhkd is definitely reasonable :+1:

**Comment by ncfavier** [comment]
Reverted my entry in `maintainers.nix` because the entry is now in `nixpkgs/master`.

**Review by sumnerevans — CHANGES_REQUESTED** [review]
I think this seems fine, but this module feels like it needs some tests.

File: .github/CODEOWNERS
Comment (ncfavier) [reply .github/CODEOWNERS]: Whoopsie. Fixed.

**Comment by ncfavier** [comment]
I don't know if there's anything interesting to test here, do you have anything in mind?

I could generate a basic configuration file and compare it against an expected result, but that seems like a lot of duplicated effort for not much gain.

File: modules/services/window-managers/bspwm/default.nix
Comment (berbiche) [inline modules/services/window-managers/bspwm/default.nix:65]: I think this was used to remove trailing whitespace since some people dislike having empty lines in their configuration file.

I'm impartial to the change.

File: modules/services/window-managers/bspwm/default.nix
Comment (berbiche) [inline modules/services/window-managers/bspwm/default.nix:75]: I opened #2107 to discuss this.

**Comment by sumnerevans** [comment]
> I could generate a basic configuration file and compare it against an expected result, but that seems like a lot of duplicated effort for not much gain.

This is what most of the tests do. It's nice to have in case somebody comes along and creates a regression.

File: modules/services/window-managers/bspwm/default.nix
Comment (berbiche) [inline modules/services/window-managers/bspwm/default.nix:25]: lists are not supported anymore?

**Review by berbiche — APPROVED** [review]
LGTM!

File: modules/services/window-managers/bspwm/default.nix
Comment (ncfavier) [reply modules/services/window-managers/bspwm/default.nix:25]: I actually missed `ignore_ewmh_fullscreen` which seems to be the only option accepting a comma-separated list of values. I'll open a new PR.

**Pull request merged**
